src/main/java/net/dries007/tfc/common/entities/predator/BigCat.java

```diff
@@ -0,0 +1,157 @@
+package net.dries007.tfc.common.entities.predator;
+
+import net.dries007.tfc.common.entities.ai.TFCBrain;
+import net.dries007.tfc.common.entities.ai.predator.PredatorAi;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.Mob;
+import net.minecraft.world.entity.Pose;
+import net.minecraft.world.entity.ai.Brain;
+import net.minecraft.world.entity.ai.attributes.AttributeSupplier;
+import net.minecraft.world.entity.ai.attributes.Attributes;
+import net.minecraft.world.entity.ai.goal.LeapAtTargetGoal;
+import net.minecraft.world.entity.ai.goal.OcelotAttackGoal;
+import net.minecraft.world.entity.ai.goal.target.NearestAttackableTargetGoal;
+import net.minecraft.world.entity.ai.memory.MemoryModuleType;
+import net.minecraft.world.entity.animal.Cat;
+import net.minecraft.world.entity.animal.Chicken;
+import net.minecraft.world.entity.animal.Pig;
+import net.minecraft.world.entity.monster.EnderMan;
+import net.minecraft.world.entity.monster.Monster;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.schedule.Activity;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.Vec3;
+import net.minecraft.world.entity.ai.goal.Goal;
+import net.minecraft.world.entity.ai.targeting.TargetingConditions;
+
+import javax.annotation.Nullable;
+import java.util.Optional;
+
+public class BigCat extends Predator {
+
+    public Vec3 location = null;
+    public Vec3 prevLocation = null;
+    public float walkProgress = 0;
+    public float walkAnimationLength = 20f; //Length of walk animation in ticks
+    public final double crouchSpeedMod;
+    public final double sprintSpeedMod;
+    public final double attackDistanceSquared;
+
+
+    public static BigCat createDiurnalCat(EntityType<? extends BigCat> type, Level level)
+    {
+        return new BigCat(type, level, true);
+    }
+
+    public BigCat(EntityType<? extends Predator> type, Level level, boolean diurnal) {
+        this(type, level, diurnal, 8, 36, 0.8, 1.5);
+    }
+    public BigCat(EntityType<? extends Predator> type, Level level, boolean diurnal, int attackAnimLength, double attackDistanceSquared, double crouchSpeedMod, double sprintSpeedMod) {
+        super(type, level, diurnal, attackAnimLength);
+        this.crouchSpeedMod = crouchSpeedMod;
+        this.sprintSpeedMod = sprintSpeedMod;
+        this.attackDistanceSquared = attackDistanceSquared;
+    }
+
+    protected void registerGoals() {
+        this.goalSelector.addGoal(7, new LeapAtTargetGoal(this, 0.4F));
+        this.goalSelector.addGoal(8, new BigCatAttackGoal(this));
+        this.targetSelector.addGoal(1, new BigCatNearestAttackableTargetGoal<>(this, Player.class, false));
+        this.targetSelector.addGoal(1, new BigCatNearestAttackableTargetGoal<>(this, Pig.class, false));
+        super.registerGoals();
```

Comment:
Yeah so we are not going to combine goal AI and brain AI. the predator implementation specifically uses Brain (see that class and PredatorAI for details) if you do this you're overriding basically all of that. So this needs to go